### PR TITLE
Progress tab cache bug fix

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [chXXXX](URL)

## Because

When DM is enabled for a facility, the cached progress tab results cause an error because they don't have any diabetes related information but the view expects it. Sentry: https://sentry.io/organizations/resolve-to-save-lives/issues/2598315876/?referrer=slack

## This addresses
Adds `diabetes_enabled?` to the cache key so the cache is updated when the DM status of a facility is changed.

### CODE REVIEW

File: app/presenters/user_analytics_presenter.rb [hunk 1]
```diff
@@ -146,7 +147,7 @@ def last_updated_at
 
   def statistics
     @statistics ||=
-      Rails.cache.fetch(statistics_cache_key, expires_in: EXPIRE_STATISTICS_CACHE_IN) {
+      Rails.cache.fetch(statistics_cache_key, expires_in: EXPIRE_STATISTICS_CACHE_IN, force: bust_cache?) {
```
Comment (prabhanshuguptagit) [inline app/presenters/user_analytics_presenter.rb:150]: Added this to help with debugging

Comment (harimohanraj89) [reply app/presenters/user_analytics_presenter.rb:150]: We also need to update the user analytics controller to _set_ `RequestStore[:bust_cache]`, right?

**Review by harimohanraj89 — APPROVED** [review]
One comment, looks good. Changing DM status constitutes a cache refresh 👍

File: app/presenters/user_analytics_presenter.rb (hunk 1 above)
Comment (prabhanshuguptagit) [reply app/presenters/user_analytics_presenter.rb:150]: Hrm right, this does not inherit from `AdminController`. 👍🏼

**Pull request merged**